fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/jobs/updateloanarrearsageing/UpdateLoanArrearsAgeingTasklet.java

```diff
@@ -67,15 +74,22 @@ public RepeatStatus execute(StepContribution contribution, ChunkContext chunkCon
         updateSqlBuilder.append(" FROM m_loan ml ");
         updateSqlBuilder.append(" INNER JOIN m_loan_repayment_schedule mr on mr.loan_id = ml.id ");
         updateSqlBuilder.append(" left join m_product_loan_recalculation_details prd on prd.product_id = ml.product_id ");
-        updateSqlBuilder.append(" WHERE ml.loan_status_id = 300 "); // active
+        updateSqlBuilder.append(" WHERE ml.loan_status_id = 300 ");// active
+        if (!loanIdsForUpdate.isEmpty()) {
+            if (!loanIdsForUpdate.equalsIgnoreCase(ALL_ACTIVE_LOANS)) {
+                updateSqlBuilder.append(" and ml.id IN (");
+                updateSqlBuilder.append(loanIdsForUpdate);
```

Comment:
Instead of directly appending it to the SQL string, let's use parameterized queries.